### ISSUE TITLE
Improved support for underlined text

### ISSUE THREAD

**Issue by piersdeseilligny** [NikolaiMe/textbox-for-pdfkit#5]
I realize that this is a known bug, however it unfortunately makes this library unuseable in a lot of use-cases. 
I'm more than happy to look into this and submit a PR, but I was just wondering if you could help give me a headstart by explaining what might be causing this issue? I looked into the source code, but didn't find anything obvious, although I noticed that `underline` is replaced by `strike` - I imagine the problem would be worse if were just underline? And is it related to an issue with pdfkit? This seems vaguely similar: https://github.com/foliojs/pdfkit/issues/994

**Comment by NikolaiMe** [comment]
Hi,

yes this is a known issue in pdfKIT. I built this library for a very specific use case and I unfortunately didn't need underline or links or strike. When I wanted to add these features I found this issue you linked above, which makes all the lines and links appear too low. And as I didn't need it for my use-case I just added a quick and dirty fix (Which is I use strike-through instead of underline).

I saw that you added a pull request, which allows users to change the basline type. So basically this is a good idea but as soon as someone is doing this and is using different text sizes or fonts (what this library is actually made for), the whole text wouldn't be aligned anymore and messed up. But yeah if we keep "alphabetic" as default but give the user the possibility to use other baselines than "alphabetic" it might help in a lot of use cases.
